src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/permissions/AdminSweeper.scala

```diff
@@ -31,54 +38,89 @@ object AdminSweeper {
+  def apply(pollAdmins: () => List[String],
+            permissionsDataSource: PermissionsDataSource,
+            initialDelay: FiniteDuration,
+            fixedRate: FiniteDuration): Behavior[Command] = {
+    Behaviors.setup { _ =>
+      Behaviors.withTimers { timers =>
+        timers.startSingleTimer(SweepSingleTimer, Sweep, initialDelay)
+        timers.startTimerAtFixedRate(SweepRateTimer, Sweep, fixedRate)
+        run(pollAdmins, permissionsDataSource)
+      }
+    }
+  }
 
-  import AdminSweeper.{Sweep, Swept}
-  implicit val executionContext: ExecutionContext = context.dispatcher
+  /**
+   * Poll for an updated list of admins, and update our user table to reflect this list.
+   * Intended to be run via a scheduler from the parent actor
+   * TODO- Implement bulk transactions for better scalability. Currently runs a DB transaction for each user whose admin status needs changing.
+   */
+  private def run(pollAdmins: () => List[String], permissionsDataSource: PermissionsDataSource): Behavior[Command] = {
 
-  override def receive: Receive = waiting
+    def waiting(): Behavior[Command] = {
+      Behaviors.receive { (context, message) =>
+        message match {
+          case Sweep =>
+            val log: Logger = context.log
+            context.pipeToSelf(synchronizeAdmins(log)(context.executionContext)) {
+              case Failure(NonFatal(nonFatal)) =>
+                log.error(s"Error synchronizing admins: ${nonFatal.getMessage}", nonFatal)
+                Swept
+              case Failure(throwable) =>
+                log.error(s"Fatal error synchronizing admins: ${throwable.getMessage}", throwable)
+                throw throwable
+              case Success(_) => Swept
+            }
+            sweeping()
+          case Swept =>
+            Behaviors.same
+        }
+      }
+    }
 
-  private def waiting: Receive = {
-    case Sweep =>
-      synchronizeAdmins andThen { case _ => self ! Swept }
-      context.become(sweeping)
-    case Swept => /* Unexpected... still ignore it. Maybe this actor restarted. */
-  }
+    def sweeping(): Behavior[Command] = {
+      Behaviors.receiveMessage {
+        case Sweep =>
+          Behaviors.same
+        case Swept =>
+          waiting()
+      }
+    }
 
-  private def sweeping: Receive = {
-    case Sweep => /* ignore, we're sweeping already */
-    case Swept => context.become(waiting)
-  }
+    def synchronizeAdmins(log: Logger)(implicit executionContext: ExecutionContext): Future[List[Int]] = {
+      // get expected and observed admins lists
+      for {
+        _ <- Future.successful(log.info(s"Beginning poll for admins"))
+        polled <- Future(pollAdmins())
+        _ <- Future.successful(log.info(s"Retrieved admins: ${polled.size}"))
+        synchronized <- synchronizeTrueAdmins(polled)
+        _ <- Future.successful(log.info(s"Synchronization adjusted admins: ${synchronized.sum}"))
+      } yield synchronized
```

Comment:
nit: could you use a variable name other than `synchronized`, since that's a keyword in Java? Java is too close to  Scala and what `synchronized` does it too close to Actor-land.